src/core/mavlink_commands.h

```diff
@@ -12,10 +12,97 @@ namespace mavsdk {
 
 class SystemImpl;
 
-class MAVLinkCommands {
+class MavlinkCommandReceiver {
 public:
-    explicit MAVLinkCommands(SystemImpl& parent);
-    ~MAVLinkCommands();
+    explicit MavlinkCommandReceiver(SystemImpl& parent);
+    ~MavlinkCommandReceiver();
+
+    struct Command {
+        uint8_t target_system_id{0};
+        uint8_t target_component_id{0};
+        uint16_t command{0};
+
+        // Int
+        MAV_FRAME frame = MAV_FRAME_GLOBAL_RELATIVE_ALT;
+        bool current = 0;
+        bool autocontinue = false;
+
+        // Long
+        uint8_t confirmation = 0;
+
+        // Mixed
+        struct Params {
+            float param1 = NAN;
+            float param2 = NAN;
+            float param3 = NAN;
+            float param4 = NAN;
+            float param5 = NAN;
+            float param6 = NAN;
+            float param7 = NAN;
+            int32_t x = 0;
+            int32_t y = 0;
+            float z = NAN;
+        } params{};
+
+        Command(mavlink_command_int_t command_int)
```

Comment:
So we now have 3 command representations? `MavlinkCommandSender::CommandLong`, `MavlinkCommandSender::CommandInt` and `MavlinkCommandReceiver::Command`? We should probably find something common, right?

Reply:
Yeah probably. Any suggestion? Maybe two parent classes, `CommandLong` and `CommandInt`?

Reply:
I have made a clear separation on the `MavlinkCommandReceiver`. We now have 4 command representations (structs), but the way they are handle are a bit different, i.e. `MavlinkCommandSender::CommandLong` is different from `MavlinkCommandReceiver::CommandLong` (one is created to send the command, the other parses a command and stores the data).